### PR TITLE
[bitnami/postgresql] Add support for autogenerated certs

### PR DESCRIPTION
**Pull request by alvneiayu**
**Description of the change**

The autogenerated certs support allow us to generate automatically certificates to test or to verify that the TLS/SSL is working properly.

**Benefits**

Make easy to test SSL scenarios.

**Checklist** 

- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. [bitnami/<name_of_the_chart>])

### CODE REVIEW

File: bitnami/postgresql/templates/_helpers.tpl [hunk 1]
```diff
@@ -310,21 +310,33 @@ postgresql: tls.enabled, volumePermissions.enabled
 Return the path to the cert file.
 */}}
 {{- define "postgresql.tlsCert" -}}
-{{- required "Certificate filename is required when TLS in enabled" .Values.tls.certFilename | printf "/opt/bitnami/postgresql/certs/%s" -}}
+{{- if and .Values.tls.enabled .Values.tls.autoGenerated }}
```
Comment (juan131) [inline bitnami/postgresql/templates/_helpers.tpl:313]: Is it necessary to check that `.Values.tls.enabled` is `true`? I thought it was implicit, I mean this macro is only invoked under `{{- if .Values.tls.enabled }}` blocks.

File: bitnami/postgresql/templates/_helpers.tpl [hunk 2]
```diff
@@ -310,21 +310,33 @@ postgresql: tls.enabled, volumePermissions.enabled
...
+{{- else -}}
+    {{- required "Certificate filename is required when TLS in enabled" .Values.tls.certFilename | printf "/opt/bitnami/postgresql/certs/%s" -}}
+{{- end -}}
 {{- end -}}
 
 {{/*
 Return the path to the cert key file.
 */}}
 {{- define "postgresql.tlsCertKey" -}}
+{{- if and .Values.tls.enabled .Values.tls.autoGenerated }}
```
Comment (juan131) [inline bitnami/postgresql/templates/_helpers.tpl:324]: Same concern here

File: bitnami/postgresql/templates/_helpers.tpl [hunk 3]
```diff
@@ -310,21 +310,33 @@ postgresql: tls.enabled, volumePermissions.enabled
...
 {{- required "Certificate Key filename is required when TLS in enabled" .Values.tls.certKeyFilename | printf "/opt/bitnami/postgresql/certs/%s" -}}
 {{- end -}}
+{{- end -}}
 
 {{/*
 Return the path to the CA cert file.
 */}}
 {{- define "postgresql.tlsCACert" -}}
-{{- printf "/opt/bitnami/postgresql/certs/%s" .Values.tls.certCAFilename -}}
+{{- if and .Values.tls.enabled .Values.tls.autoGenerated }}
```
Comment (juan131) [inline bitnami/postgresql/templates/_helpers.tpl:335]: Same concern here

File: bitnami/postgresql/templates/_helpers.tpl [hunk 4]
```diff
@@ -335,3 +347,25 @@ Return the path to the CRL file.
...
+{{- if and .Values.tls.enabled .Values.tls.autoGenerated (not .Values.tls.certificatesSecret) }}
+    {{- true -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Return the path to the CA cert file.
+*/}}
+{{- define "postgresql.tlsSecretName" -}}
+{{- if .Values.tls.enabled }}
```
Comment (juan131) [inline bitnami/postgresql/templates/_helpers.tpl:364]: Same concern here

File: bitnami/postgresql/templates/tls-secrets.yaml [hunk 5]
```diff
@@ -0,0 +1,26 @@
+{{- if (include "postgresql.createTlsSecret" . )}}
+{{- $ca := genCA "postgresql-internal-ca" 365 }}
```
Comment (juan131) [inline bitnami/postgresql/templates/tls-secrets.yaml:2]: Just curious why the `-internal-` affix?

File: bitnami/postgresql/templates/tls-secrets.yaml [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+{{- $fullname := include "common.names.fullname" . }}
+{{- $releaseNamespace := .Release.Namespace }}
+{{- $clusterDomain := .Values.clusterDomain }}
+{{- $serviceName := include "common.names.fullname" . }}
```
Comment (juan131) [inline bitnami/postgresql/templates/tls-secrets.yaml:6]: Both `$fullname` and `$serviceName` variables are assigned the same value. Remove one in favor or the other.

File: bitnami/postgresql/templates/tls-secrets.yaml [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+{{- $headlessServiceName := printf "%s-headless" (include "common.names.fullname" .) }}
+{{- $altNames := list (printf "*.%s.%s.svc.%s" $serviceName $releaseNamespace $clusterDomain) (printf "%s.%s.svc.%s" $serviceName $releaseNamespace $clusterDomain) (printf "*.%s.%s.svc.%s" $headlessServiceName $releaseNamespace $clusterDomain) (printf "%s.%s.svc.%s" $headlessServiceName $releaseNamespace $clusterDomain) $fullname }}
+{{- $crt := genSignedCert $fullname nil $altNames 365 $ca }}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ include "common.names.fullname" . }}-crt
```
Comment (juan131) [inline bitnami/postgresql/templates/tls-secrets.yaml:13]: I prefer "printf" for this, but do as you prefer:
```suggestion
  name: {{ printf "%s-crt" (include "common.names.fullname" .) }}
```

**Review by juan131 — CHANGES_REQUESTED** [review]
Thanks for the PR! Some minor requests

File: bitnami/postgresql/templates/tls-secrets.yaml (hunk 5 above)
Comment (alvneiayu) [reply bitnami/postgresql/templates/tls-secrets.yaml:2]: I added this affix to say that it is a certificate to use internally (not in production) but looking the decision with perspective, it is not the best name. I changed it.

**Review by juan131 — APPROVED** [review]
LGTM! Thanks for applying the suggestions

**Pull request merged**